packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -619,6 +771,82 @@ enum SelectionStatus {
   none,
 }
 
+/// The details of a selection.
+///
+/// This object is created by callers of the [SelectionListenerSelectionChangedCallback] callback.
+///
+/// This includes information such as the status of the selection
+/// indicating if it is collapsed or uncollapsed, the start and end offsets
+/// of the selection local to the [SelectionListener] that reports this object,
+/// and whether the selection is ongoing.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides a [SelectionDetails] object
+///     for the selection under its subtree in its [SelectionListener.onSelectionChanged]
+///     callback.
+@immutable
+class SelectionDetails {
```

Comment:
is the SelectionDetails meant to be the public facing API? if so we should try to make getSelectionList protected and SelectedContentRange internal

Reply:
By internal do you mean private?